### PR TITLE
parser: support sql_mode `high_not_precedence`

### PR DESCRIPTION
**Pull request by spongedu**
for #4768 , support sql_mode `high_not_precedence`

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -790,7 +792,7 @@ import (
 %left 	'*' '/' '%' div mod
 %left 	'^'
 %left 	'~' neg
-%right 	not
+%right 	not not2
```
Comment (zz-jason) [inline parser/parser.y]: can we change `not2` to another more specified name ?

Comment (spongedu) [reply parser/parser.y]: @zz-jason  MySQL defines token like this. I think it's better to keep consistent with MySQL.

Comment (zz-jason) [reply parser/parser.y]: ok

File: parser/parser.y [hunk 2]
```diff
@@ -2763,6 +2765,10 @@ SimpleExpr:
 	{
 		$$ = &ast.UnaryOperationExpr{Op: opcode.Plus, V: $2}
 	}
+| not2 SimpleExpr %prec neg
```
Comment (coocood) [inline parser/parser.y]: `parser.y` use tab for alignment.

File: parser/parser.y [hunk 3]
```diff
@@ -2763,6 +2765,10 @@ SimpleExpr:
 	{
 		$$ = &ast.UnaryOperationExpr{Op: opcode.Plus, V: $2}
 	}
+|	not2 SimpleExpr %prec neg
+    {
```
Comment (coocood) [inline parser/parser.y]: alignment

File: parser/parser.y [hunk 4]
```diff
@@ -2763,6 +2765,10 @@ SimpleExpr:
...
+		$$ = &ast.UnaryOperationExpr{Op: opcode.Not, V: $2}
+    }
```
Comment (coocood) [inline parser/parser.y]: ditto

**Comment by coocood** [comment]
LGTM

**Pull request merged**